yt_dlp/extractor/ximalaya.py

```diff
@@ -71,23 +81,92 @@ class XimalayaIE(XimalayaBaseIE):
                 'like_count': int,
             },
         },
+        {
+            # VIP-restricted audio
+            'url': 'https://www.ximalaya.com/sound/562111701',
+            'only_matching': True,
+        },
     ]
 
+    @staticmethod
+    def _decrypt_filename(file_id, seed):
+        cgstr = ''
+        key = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ/\\:._-1234567890'
+        for _ in range(len(key)):
+            seed = float(int(211 * seed + 30031) % 65536)
+            r = int(seed / 65536 * len(key))
+            cgstr += key[r]
+            key = key.replace(key[r], '')
+        parts = file_id.split('*')
+        filename = ''.join(cgstr[int(part)] for part in parts if part.isdigit())
```

Comment:
I think `isdecimal` is what we want here
```suggestion
        filename = ''.join(cgstr[int(part)] for part in parts if part.isdecimal())
```

Reply:
Here is a example of `file_id`, I think `isdigit()` might be sufficient:

```json
{
  "ret": 0,
  "msg": "0",
  "trackId": 749837049,
  "uid": 37504890,
  "albumId": 83105493,
  "title": "第1集 雷旭",
  "domain": "http://audiopay.cos.tx.xmcdn.com",
  "totalLength": 4833714,
  "sampleDuration": 180,
  "sampleLength": 1156925,
  "isAuthorized": true,
  "apiVersion": "1.0.0",
  "seed": 9967,
  "fileId": "40*42*2*17*28*1*32*40*33*67*7*21*28*35*28*54*25*31*2*50*28*38*33*18*9*33*18*6*33*45*13*51*52*14*64*14*13*24*61*3*38*6*18*8*65*40*1*41*47*29*65*22*7*12*24*21*28*",
  "buyKey": "FM",
  "duration": 792,
  "ep": "qz91a9wrGyYFgfkIb9tN5FsKBElygSTGTdUVfA3mC377mb6Q3/+Bm/AP1f2J0KdhXeJ2",
  "highestQualityLevel": 2,
  "downloadQualityLevel": 1,
  "authorizedType": 1,
  "volumeGain": -5.4
}
```

Reply:
IMO it's always better to be more specific, not less. `isdecimal` is a subset of `isdigit`

Reply:
got it~